config-store-api/src/main/java/io/stargate/config/store/api/ConfigWithOverrides.java

```diff
@@ -36,6 +39,84 @@ public ConfigWithOverrides(@Nonnull Map<String, Object> configMap, String module
     return configMap;
   }
 
+  /**
+   * It retrieves the list of values from underlying configMap checking for its presence and the
+   * type. If the value for a given settingName does not have a value, it throws an {@code
+   * IllegalArgumentException}. If the type of the setting value is not a {@link List} it throws
+   * {@code IllegalArgumentException}. If the underlying type of any element in this list does not
+   * match the expectedType, it throws {@code IllegalArgumentException}.
+   *
+   * @param settingName - it will be used as a key of underlying configMap.
+   * @param expectedType - it will be used to check if the type of an actual list element value
+   *     matches.
+   * @return the list of values associated with settingName. Each value matches expectedType.
+   */
+  @Nonnull
+  @SuppressWarnings("unchecked")
+  public <T> List<T> getSettingValueList(String settingName, Class<T> expectedType) {
```

Comment:
I think that methods for getting a list and raw types is enough for us at this point.
The YAML format supports only lists and maps as collection types. Currently, we don't have a use case for the map.
We could create an extensive API that uses guava's `TypeToken,` but:
- it's better to not add the guava dependency in the `API` module
- currently, we don't need more types

Reply:
Is this method just a convenience method for `getSettingValue("foo", List.class)` or is this to make sure it's a list of strings rather than a list of integers?